File: app/src/main/java/net/kibotu/android/recyclerviewpresenter/app/FakeDataGenerator.java
package net.kibotu.android.recyclerviewpresenter.app;

import java.util.Random;

import static java.text.MessageFormat.format;

/**
 * Created by <a href="https://about.me/janrabe">Jan Rabe</a>.
 */

class FakeDataGenerator {

    static String createRandomImageUrl() {
        boolean landscape = new Random().nextBoolean();
        return format("https://picsum.photos/{0}/{1}/", landscape ? 400 : 200, landscape ? 200 : 400);
    }
}

File: app/src/main/java/net/kibotu/android/recyclerviewpresenter/app/FakeDataGenerator.kt
package net.kibotu.android.recyclerviewpresenter.app

import com.exozet.android.core.utils.MathExtensions
import java.text.MessageFormat.format
import java.util.*

/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */

internal object FakeDataGenerator {

    fun createRandomImageUrl(): String {
        val random = Random()
        val landscape = random.nextBoolean()
        val endpoint = random.nextBoolean()

        val width = MathExtensions.random(400, 300)
        val height = MathExtensions.random(300, 200)

        return format(
            if (endpoint)
                "https://lorempixel.com/{0}/{1}/"
            else
                "https://picsum.photos/{0}/{1}/",
            if (landscape) width else height, if (landscape) height else width
        )
    }
}
